pubsub/aws/snssqs/metadata.go

```diff
@@ -139,11 +141,25 @@ func (s *snsSqs) getSnsSqsMetatdata(metadata pubsub.Metadata) (*snsSqsMetadata,
 		return nil, err
 	}
 
+	if err := md.setConcurrencyMode(props); err != nil {
+		return nil, err
+	}
+
 	s.logger.Debug(md.hideDebugPrintedCredentials())
 
 	return md, nil
 }
 
+func (md *snsSqsMetadata) setConcurrencyMode(props map[string]string) error {
+	c, err := pubsub.Concurrency(props)
+	if err != nil {
+		return err
+	}
+	md.concurrencyMode = c
```

Comment:
Yes. That's true for every component using that pubsub metadata field. I mean that deciding that "parallel" is the default is the project decision. You think we should change that? Document that?

Reply:
Does this concurrency metadata field affect multiple (all?) pub/sub components?

Reply:
No this only affects this component. It however uses a helper function that scans metadata to look for a concurrency value and returns "parallel" if it can't find one.

The problem is that before this change the SNS/SQS component will likely have used a different default concurrency value.

Maybe we should just duplicate the helper function here within the component, but without specifying a default value so we can retain the existing behavior (whatever that even is).

Reply:
The existing behavior is very clearly single concurrency mode.

Reply:
AFAIR the Rabbit MQ component also uses this construct. I'm totally fine with setting the "single" mode as default although I'm not too sure if that's what most users would expect (we were surprised at first) because most webservers and languages are concurrent out of the box nowadays. I also want to remind that one of the advantages of SQS is batching of message consumption and therefore sending the batch concurrently to the handler fits the idea nicely

Reply:
I think we should enable `parallel` as the default for SQS and explicitly mention it on the documentation (and also as a highlighted change in the 1.8 release notes). As a Beta component we can make this change now, also this doesn't change the APIs so I feel comfortable with this.